tests/integration/telemetry/test_experimental_metric.py

```diff
@@ -152,8 +150,11 @@ def test_must_send_cdk_project_type_metrics(self):
             process.communicate()
 
             all_requests = server.get_all_requests()
-            self.assertEqual(1, len(all_requests), "Command run metric must be sent")
-            request = all_requests[0]
+            self.assertGreaterEqual(len(all_requests), 1, "Command run metric must be sent")
+            request = None
+            for req in all_requests:
+                print(req["data"])
+                request = req if "commandRun" in req["data"]["metrics"][0].keys() else all_requests[-1]
```

Comment:
Are we only looking for the `req` in `all_requests` with the `commandRun` metric? I don't understand why we need `else all_requests[-1]`. Looks like `request` is assigned as `all_requests[-1]` regardless

Reply:
Yes we only want the `commandRun` metric for this test. Initially I wanted to check which of the metrics had `commandRun` in the name, so I tried iterating through all of them. However on second look, I can see that the `commandRun` metric will be guaranteed as the last metric thanks to the ordering of `EventTracker.send_events(); telemetry.emit(metric)`. Will fix this in the next commit. Good catch!